src/oop/static_modifier/hard/q3/MainTest.java

```diff
@@ -0,0 +1,152 @@
+                Arguments.of("SugarShack", initialGummies / 2),
+                Arguments.of("SweetSpot", initialGummies - 75),
+                Arguments.of("TreatTown", initialGummies / 3),
+                Arguments.of("BonbonBoulevard", initialGummies - 60),
+                Arguments.of("Lollipop Lane", initialGummies / 4),
+                Arguments.of("Jellybean Junction", initialGummies - 15),
+                Arguments.of("Confectionery Corner", initialGummies / 5),
+                Arguments.of("Dulce Drive", initialGummies - 90)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent3")
+    @Order(4)
+    public void lightbulbConstructorInitializesValuesCorrectly(String value1) throws Throwable {
+        String wrongValueMessage = "The %s constructor did not initialize the %s attribute to the correct value based on the parameters passed to the constructor.";
+        Object[][] constructorArgs = {
+                {value1, String.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName), String.format(wrongValueMessage, className, attributeName));
+    }
+
+    @Test
+    @Order(5)
+    public void awardIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName, new Class<?>[]{int.class}), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(testClass.hasModifier(methodName, new Class<?>[]{int.class}, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(testClass.hasReturnType(methodName, new Class<?>[]{int.class}, Void.TYPE), String.format(incorrectReturnType, methodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    @Order(6)
+    public void candyStoreClassHasCorrectAwardMethod(String value1, int value2) throws Throwable {
+        Object[][] constructorArgs = {
+                {value1, String.class}
+        };
+        Object classInstance = testClass.createInstance(constructorArgs);
+        Clause[] testSentence = new Clause[]{
+                new StringLiteral("There are " + (initialGummies - value2) + " more to give away after the " + value1 + " award.")
+        };
+        Object[][] arguments = {
+                {value2, int.class}
+        };
+        testClass.callMethod(methodName, arguments, classInstance, testSentence);
+        testClass.setFieldValue(null, initialGummies, staticAttributeName);
+    }
+
```

Comment:
You should also add a test where the award method is called multiple times on the same object to ensure everything is being updated properly.